### PR TITLE
Add documentation on how to migrate away from a retracted version

### PR DESCRIPTION
**Pull request by szakarias**
Partly fixes  https://github.com/dart-lang/pub/issues/4033

### CODE REVIEW

**Review by sigurdm — APPROVED** [review]
LGTM

File: src/tools/pub/publishing.md [hunk 1]
```diff
@@ -409,6 +409,43 @@ the dependent package must pin the version in the
...
+If there is no newer version available, the best action might be to downgrade to
+the newest non-retracted version. There are two ways to get this version.
+
+The first way is by using [pub tool](/tools/pub/cmd) commands: Run `dart pub
+downgrade <package>` and then run `dart pub upgrade <package>`. The `downgrade`
+command will get the lowest version of `<package>` matching the dependency
+constraint in the `pubspec.yaml` file. The following `upgrade` command will then
+get the newest non-retracted version available.
+
+The second way is by editting the `pubspec.lock` file manually. Delete the
```
Comment (MaryaBelanger) [inline src/tools/pub/publishing.md:434]:
```suggestion
The second way is by editing the `pubspec.lock` file manually. Delete the
```

**Review by MaryaBelanger — APPROVED** [review]
Just a typo, otherwise LGTM. Thanks pub team!

File: src/tools/pub/publishing.md [hunk 2]
```diff
@@ -409,6 +409,43 @@ the dependent package must pin the version in the
...
+
+The first way is by using [pub tool](/tools/pub/cmd) commands: Run `dart pub
+downgrade <package>` and then run `dart pub upgrade <package>`. The `downgrade`
+command will get the lowest version of `<package>` matching the dependency
+constraint in the `pubspec.yaml` file. The following `upgrade` command will then
+get the newest non-retracted version available.
+
+The second way is by editing the `pubspec.lock` file manually. Delete the
+entire package entry for the package with the retracted version and then run
+`dart pub get`.
```
Comment (parlough) [inline src/tools/pub/publishing.md:436]: It was a bit hard for me to parse the first paragraph. What do you think about breaking these up into steps?

```suggestion
The first way is by using [pub tool](/tools/pub/cmd) commands:

1. Run `dart pub downgrade <package>` to
   get the lowest version  of the specified package that
   matches the constraints in the `pubspec.yaml` file.
2. Run `dart pub upgrade <package>` to get the
   newest compatible and non-retracted version available.

The second way is by editing the `pubspec.lock` file manually:

1. Delete the entire package entry for the package with the retracted version.
2. Run `dart pub get` to get the
   newest compatible and non-retracted version available.
```

**Review by parlough — APPROVED** [review]
Thanks @szakarias, it's nice to have how to resolve this situation documented more clearly!

I have one potential suggestion to consider, otherwise merge when you're ready. If you aren't able to merge, let me know, and I can merge it. Thanks!

**Comment by szakarias** [comment]
Thanks @parlough for the comment. It would be great if you can commit for me. :-)

**Pull request merged**